cumulusci/tasks/salesforce/PackageUpload.py

```diff
@@ -116,6 +111,35 @@ def _run_task(self):
                 )
             )
 
+    def _handle_package_upload_error(self):
+        self.logger.error("Package upload failed with the following errors")
+        error = {"message": ""}
+        apex_test_failures = False
+        for error in self.upload["Errors"]["errors"]:
+            self.logger.error("  {}".format(error["message"]))
+            if error["message"] == "ApexTestFailure":
+                apex_test_failures = True
+
+        if apex_test_failures:
+            self.logger.error("Failed Apex Tests:")
+            test_start_datetime = datetime.utcnow() - timedelta(
+                seconds=self.upload_time_seconds
+            )
+            failed_tests_query = f"SELECT ApexClass.Name, MethodName, Message FROM ApexTestResult WHERE Outcome='Fail' AND TestTimestamp > {test_start_datetime.isoformat()+'Z'}"
+            results = self.tooling.query(failed_tests_query)
+            for test in results["records"]:
+                self.logger.error(
+                    f"    {test['ApexClass']['Name']}.{test['MethodName']}"
```

Comment:
Since we're going to need the `StackTrace`, you may want to consider just dumping these results into a table.